### ISSUE TITLE
Android: apksigner not found with android-sdk package from Ubuntu Universe (older SDK)

### ISSUE THREAD

**Issue by snoopdouglas** [godotengine/godot#45844]
**Godot version:**
3.2.4-rc1

**OS/device including version:**
Ubuntu Linux 20.04
OpenJDK 11.0.9.1
Android SDK 25

**Issue description:**
After fixing the editor config as per #45341, I noticed that the editor still couldn't find apksigner, and it showed an error when I went to export.

I found this could be fixed by softlinking `apksigner` to `apksigner.jar` in `/usr/lib/android-sdk/build-tools/27.0.1`.

I believe this wasn't an issue in 3.2.3 because the apksigner path needed to be set explicitly; I'd set it as `/usr/bin/apksigner` (which, incidentally, links to `/usr/lib/android-sdk/build-tools/27.0.1/apksigner.jar`).

**Steps to reproduce:**
* Run Godot 3.2.4-rc1 on Ubuntu 20.04
* Install AdoptJDK and Android SDK packages as per versions above
* Set editor SDK path to `/usr/lib/android-sdk`
* Observe that errors are shown until linking `apksigner` to `apksigner.jar`

**Minimal reproduction project:**
n/a

**Comment by Calinou** [comment]
cc @m4gr3d

**Comment by akien-mga** [comment]
How do you install the Android SDK? Any specific reason why it's using `build-tools` 27.0.1 instead of a newer version? Godot 3.2 uses 29.0.x. Might be that they renamed the jar in later versions.

**Comment by snoopdouglas** [comment]
I installed the `android-sdk` package via apt. I'm aware the docs recommend downloading Android Studio directly, but I didn't see anything regarding minimum versions.

**Comment by akien-mga** [comment]
I'm not sure the `android-sdk` package of Ubuntu Universe would supported, it's not exactly the same thing as the Android SDK you can get from Google (I'm not even sure how Ubuntu can redistribute it as the SDK license prohibits it IINM...).

If all that's needed is to add a check for `.jar` extension as a fallback, we can likely support it, but you might run into other issues using that version. It seems that they stopped updating it to match upstream changes, for example they provide the API 23 (Android 6.0) platform SDK: https://packages.ubuntu.com/hirsute/android-sdk

Note that you don't have to install Android Studio, you can just install the command line tools and use `sdkmanager` to set up the latest `build-tools` and `platform-tools` to have `apksigner` and `adb`.

**Comment by snoopdouglas** [comment]
Thanks - I've just confirmed this works without a hitch with the 3.2.4 docs' recommended `sdkmanager` install.

**Comment by m4gr3d** [comment]
@snoopdouglas Sounds good! Seems this issue can be closed then since the docs already describe the process for grabbing the sdk. 

@Calinou @akien-mga do you think we should add a warning not to use the distro packaged Android sdk?

**Comment by akien-mga** [comment]
> do you think we should add a warning not to use the distro packaged Android sdk?

I guess that might be sufficient yes, as I don't know of many distros actively packaging the Android SDK aside from Debian (and thus Ubuntu), and by the time a Debian release is published, its packaged SDK is useless as outdated.

**Comment by Calinou** [comment]
> (I'm not even sure how Ubuntu can redistribute it as the SDK license prohibits it IINM...).

The Ubuntu package is pretty much the same as the Debian package – it's a version with all components rebuilt from source and no Google binaries unused. Unfortunately, it also gets outdated quickly.

**Comment by m4gr3d** [comment]
Thanks @Calinou for updating the docs!

**Comment by knkp** [comment]
So, in case anyone is interested I found a quick way to make it work for the android sdk installed from apt.

After you have completed the installation just run the following:
```
cd /usr/lib/android-sdk/build-tools/debian
sudo ln -s apksigner.jar apksigner
```

This resolves the only issue I found and upon completing the "Dodge The Creeps" tutorial, exporting it and installing it on my phone went exactly as expected.

I'm on 
Ubuntu 20.04
kernel: 5.11.0-36-generic

Not sure if it matters, but I hope someone found that helpful anyway.
